test/unit/CowWrapperHelpers.t.sol

```diff
@@ -0,0 +1,266 @@
+        );
+        helpers.verifyAndBuildWrapperData(wrapperCalls);
+    }
+
+    function test_verifyAndBuildWrapperData_RevertsOnWrapperDataNotFullyConsumed() public {
+        CowWrapperHelpers.WrapperCall[] memory wrapperCalls = new CowWrapperHelpers.WrapperCall[](1);
+        wrapperCalls[0] = CowWrapperHelpers.WrapperCall({
+            target: address(wrapper1), // Consumes 4 bytes
+            data: hex"deadbeefcafe" // 6 bytes, but wrapper only consumes 4
+        });
+
+        vm.expectRevert(abi.encodeWithSelector(CowWrapperHelpers.WrapperDataNotFullyConsumed.selector, 0, hex"cafe"));
+        helpers.verifyAndBuildWrapperData(wrapperCalls);
+    }
+
+    function test_verifyAndBuildWrapperData_RevertsOnWrapperDataMalformed() public {
+        CowWrapperHelpers.WrapperCall[] memory wrapperCalls = new CowWrapperHelpers.WrapperCall[](1);
+        wrapperCalls[0] = CowWrapperHelpers.WrapperCall({target: address(wrapper1), data: hex"deadbeef"});
+
+        bytes memory errorData = hex"feab";
+        vm.mockCallRevert(address(wrapper1), abi.encodeWithSelector(wrapper1.parseWrapperData.selector), errorData);
+
+        vm.expectRevert(abi.encodeWithSelector(CowWrapperHelpers.WrapperDataMalformed.selector, 0, errorData));
+        helpers.verifyAndBuildWrapperData(wrapperCalls);
+    }
+
+    function test_verifyAndBuildWrapperData_RevertsOnSettlementContractShouldNotBeSolver() public {
+        // Add settlement as a solver (which should not be allowed)
+        solverAuth.setSolver(address(mockSettlement), true);
+
+        CowWrapperHelpers.WrapperCall[] memory wrapperCalls = new CowWrapperHelpers.WrapperCall[](1);
+        wrapperCalls[0] = CowWrapperHelpers.WrapperCall({target: address(wrapper1), data: hex"deadbeef"});
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                CowWrapperHelpers.SettlementContractShouldNotBeSolver.selector,
+                address(mockSettlement),
+                address(solverAuth)
+            )
+        );
+        helpers.verifyAndBuildWrapperData(wrapperCalls);
+    }
+
+    function test_verifyAndBuildWrapperData_EmptyWrapperData() public view {
+        CowWrapperHelpers.WrapperCall[] memory wrapperCalls = new CowWrapperHelpers.WrapperCall[](2);
+        wrapperCalls[0] = CowWrapperHelpers.WrapperCall({
+            target: address(wrapper3), // Consumes 0 bytes
+            data: hex""
+        });
+        wrapperCalls[1] = CowWrapperHelpers.WrapperCall({
+            target: address(wrapper3), // Consumes 0 bytes
+            data: hex""
+        });
+
+        bytes memory result = helpers.verifyAndBuildWrapperData(wrapperCalls);
+
+        // Should contain: data[0] + target[1] + data[1]
+        bytes memory expected = abi.encodePacked(uint16(0), hex"", address(wrapper3), uint16(0), hex"");
+        assertEq(result, expected);
+    }
+
+    function test_verifyAndBuildWrapperData_MixedWrapperDataSizes() public view {
```

Comment:
This test seems to be exactly the same as `test_verifyAndBuildWrapperData_MultipleWrappers` and could be dropped.